kernel/dev/vga/graphics.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file graphics.cpp
+ * @author Keeton Feavel (keetonfeavel@cedarville.edu)
+ * @brief
+ * @version 0.1
+ * @date 2021-06-11
+ *
+ * @copyright Copyright the Panix Contributors (c) 2021
+ *
+ */
+#include <dev/vga/graphics.hpp>
+#include <dev/vga/fb.hpp>
+// Types
+#include <stddef.h>
+#include <stdint.h>
+// Memory management & paging
+#include <mem/heap.hpp>
+#include <mem/paging.hpp>
+// System library functions
+#include <sys/panic.hpp>
+#include <lib/assert.hpp>
+#include <lib/stdio.hpp>
+// Bootloader
+#include <boot/Handoff.hpp>
+// Debug
+#include <dev/serial/rs232.hpp>
+
+namespace fb {
+
+FramebufferInfo fbInfo;
+// Cached values
+static void* addr = NULL;
+static uint32_t width = 0;
+static uint32_t height = 0;
+static uint16_t depth = 0;
+static uint32_t pitch = 0;
+static uint32_t pixelwidth = 0;
+static uint8_t r_size = 0;
+static uint8_t r_shift = 0;
+static uint8_t g_size = 0;
+static uint8_t g_shift = 0;
+static uint8_t b_size = 0;
+static uint8_t b_shift = 0;
+static bool initialized = false;
+
+bool isInitialized() { return initialized; }
+
+static void testPattern() {
+    const uint32_t BAR_COLOUR[8] =
```

Comment:
I'd prefer to spell color the American way, but I don't care that much if you want to keep it like that.